pkg/pipeline/CiCdPipelineOrchestrator.go

```diff
@@ -1307,6 +1343,12 @@ func (impl CiCdPipelineOrchestratorImpl) GetCdPipelinesForApp(appId int) (cdPipe
 
 	var pipelines []*bean.CDPipelineConfigObject
 	for _, dbPipeline := range dbPipelines {
+		preDeployStage, postDeployStage, err := impl.pipelineStageService.GetCdPipelineStageDataDeepCopy(dbPipeline.Id)
```

Comment:
we should create a new API and accepting request for multiple pipelineIds. that would be faster